src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -23,26 +23,69 @@ public static class EventHubExceptionTelemetryProcessor
+
+            logger.LogError(ex);
 
             if (shouldLogMetric)
             {
-                logger.LogMetric(EventMetrics.HandledException(errorMetricName ?? GetErrorMetricName(exception), ConnectorOperation.Setup), 1);
+                if (ex.Equals(exception))
+                {
+                    logger.LogMetric(
+                        EventMetrics.HandledException(
+                            errorMetricName ?? $"{ErrorType.EventHubError}{EventHubErrorCode.GeneralError}",
+                            ConnectorOperation.Setup),
+                        1);
+                }
+                else
+                {
+                    var processor = new EventHubConfigurationExceptionTelemetryProcessor();
+                    processor.HandleException(ex, logger, ConnectorOperation.Setup);
+                }
             }
         }
 
-        private static string GetErrorMetricName(Exception exception)
+        public static Exception CustomizeException(Exception exception)
         {
+            string message = exception.Message;
+            string helpLink = exception.HelpLink;
+
             switch (exception)
             {
                 case EventHubsException _:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.OperationError}";
+                    EventHubsException eventHubsException = (EventHubsException)exception;
+                    switch (eventHubsException.Reason)
+                    {
+                        case EventHubsException.FailureReason.ConsumerDisconnected:
+                            message = "Verify that the provided Event Hub is not already receiving data for another IoT connector or Azure resource.";
+                            break;
+                        case EventHubsException.FailureReason.ResourceNotFound:
+                            message = "Verify that the provided Event Hubs Namespace contains the provided Event Hub and that the provided Event Hub contains the provided consumer group.";
+                            break;
+                        case EventHubsException.FailureReason.ServiceCommunicationProblem:
+                            message = "Verify that the provided Event Hub Namespace, Event Hub name, and consumer group are correct and that access permissions to the provided Event Hub have been granted.";
+                            break;
+                        default:
+                            return exception;
+                    }
+
+                    return new InvalidEventHubException(message, exception, helpLink, EventHubErrorCode.InstanceError.ToString());
+
+                case InvalidOperationException _:
+                    message = "Verify that the provided Event Hub contains the provided consumer group.";
+                    return new InvalidEventHubException(message, exception, helpLink, EventHubErrorCode.InstanceError.ToString());
+
                 case SocketException _:
```

Comment:
Is there a SocketErrorCode we can also check against for this case. 
This might not be the only scenario when we will get a socket exception.

Reply:
Thanks for noticing this. Will only handle for `SocketError.HostNotFound`.